certbot-nginx/certbot_nginx/_internal/http_01.py

```diff
@@ -113,12 +116,12 @@ def _mod_config(self):
         with io.open(self.challenge_conf, "w", encoding="utf-8") as new_conf:
             nginxparser.dump(config, new_conf)
 
-    def _default_listen_addresses(self):
+    def _default_listen_addresses(self) -> List[Addr]:
         """Finds addresses for a challenge block to listen on.
-        :returns: list of :class:`certbot_nginx._internal.obj.Addr` to apply
+        :returns: list of :class:`certbot_nginx._internal.Addr` to apply
         :rtype: list
         """
-        addresses: List[obj.Addr] = []
+        addresses: List[Optional[Addr]] = []
```

Comment:
Because it turns out that the Nginx implementation of `Addr.fromstring()` can return `None`: 
```
        # Ignore UNIX-domain sockets
        if addr.startswith('unix:'):
            return None
```

So I propagate this information accordingly to callers.